wcfsetup/install/files/lib/system/template/TemplateScriptingCompiler.class.php

```diff
@@ -689,19 +689,27 @@ protected function compileForeachTag($foreachTag) {
 			$foreachProp = "\$this->v['tpl']['foreach'][".$args['name']."]";
 		}
 		
+		$hash = StringUtil::getRandomID();
+		
 		$phpCode = "<?php\n";
+		$phpCode .= "\$_foreach_".$hash." = ".$args['from'].";\n";
+		$phpCode .= "\$_foreach_".$hash."_cnt = 1;\n";
```

Comment:
This will effectively evaluate to `1` for `NULL` values, but `count()` explicitly evaluates to `0` for this special value.